src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -141,4 +179,123 @@
+        NSDateComponents *startDateComponents = [[NSDateComponents alloc] init];
+        startDateComponents.era = era;
+        // set the date to Jan 1, 1
+        startDateComponents.month = 1;
+        startDateComponents.day = 1;
+        startDateComponents.year = 1;
+        NSDate *date = [japaneseCalendar dateFromComponents:startDateComponents];
+        NSDate *startDate = date;
+        int32_t currentEra;
+
+        for (int month = 0; month <= 12; month++)
+        {
+            NSDateComponents *eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+            currentEra = [eraComponents era];
+            if (currentEra == era)
+            {
+                for (int day = 0; day < 31; day++)
+                {
+                    // subtract 1 day at a time until we get out of the specified Era
+                    startDateComponents.day = startDateComponents.day - 1;
+                    date = [japaneseCalendar dateFromComponents:startDateComponents];
+                    eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+                    currentEra = [eraComponents era];
+                    if (currentEra != era)
+                    {
+                        // add back 1 day to get back into the specified Era
+                        startDateComponents.day = startDateComponents.day + 1;
+                        startDate = [japaneseCalendar dateFromComponents:startDateComponents];
+                        NSCalendar *gregorianCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+                        NSDateComponents *components = [gregorianCalendar components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:startDate];
+                        *startYear = [components year];
+                        *startMonth = [components month];
+                        *startDay = [components day];
+                        return 1;
+                    }
+                }
+            }
+            // add 1 month at a time until we get into the specified Era
+            startDateComponents.month = startDateComponents.month + 1;
+            date = [japaneseCalendar dateFromComponents:startDateComponents];
+            eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+            currentEra = [eraComponents era];
+        }
+
+        return 0;
+    }
+}
+
+/*
+Function:
+GetCalendarsNative
+
+Returns the list of CalendarIds that are available for the specified locale.
+*/
+int32_t GlobalizationNative_GetCalendarsNative(const char* localeName, CalendarId* calendars, int32_t calendarsCapacity)
+{
+    @autoreleasepool
+    {
+        NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+        NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
```

Comment:
This doesn't seem to have an effect on the following code. It feels like the calendars in `calendarIdentifiers` should depend on the current locale.

Reply:
I didn't find way to retrieve a list of all supported calendars for a specific locale in Objective-C, as iOS/macOS doesn't expose a direct API for this purpose. That is why I decided to use predefined list of commonly used calendars but forgot to remove locale related parameters, will update.